api/src/main/java/org/eclipse/microprofile/jwt/algorithm/KeyEncryptionAlgorithm.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2016-2017 Contributors to the Eclipse Foundation
+ *
+ *  See the NOTICE file(s) distributed with this work for additional
+ *  information regarding copyright ownership.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  You may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.eclipse.microprofile.jwt.algorithm;
+
+/**
+ * JWT JSON Web Key Encryption (Management) Algorithms.
+ * 
+ * @see <a href="https://tools.ietf.org/html/rfc7518#section-4">https://tools.ietf.org/html/rfc7518#section-4</a>
+ */
+public enum KeyEncryptionAlgorithm {
+    RSA_OAEP("RSA-OAEP"),
+    RSA_OAEP_256("RSA-OAEP-256"),
+    ECDH_ES("ECDH-ES"),
+    ECDH_ES_A256KW("ECDH-ES+A256KW"),
+    A256KW("A256KW");
+
+    private String algorithmName;
+
+    private KeyEncryptionAlgorithm(String algorithmName) {
+        this.algorithmName = algorithmName;
+    }
+
+    public String getAlgorithm() {
+        return algorithmName;
+    }
+
+    public static KeyEncryptionAlgorithm fromAlgorithm(String algorithmName) {
+        return KeyEncryptionAlgorithm.valueOf(algorithmName.replaceAll("-", "_").replaceAll("\\+", "_"));
```

Comment:
This logic fails when passing a parameter which doesn't correspond to valid valid or when null is passed.

This is much more safer

return Arrays.stream(KeyEncryptionAlgorithm.values()).filter(e -> e.algorithmName.equals(algorithmName))
                .findAny().orElse(null);

Reply:
@rdebusscher I've looked more carefully at this comment and I don't think it is a good idea, sorry about it. For few reasons: several of my colleagues have noted that the streaming API is expensive and it simply looks as if it is too much for the propose of converting an algorithm like `RSA-OAEP` into a valid enum value like `RSA_OAEP` which is the reason behind this code which is your code wold not do.
I'll make sure that NPE is avoided. And I prefer the exception thrown if no valid enum values exists

Reply:
ok, But String manipulations like replaceAll are also expensive. But fine for me.